### ISSUE TITLE
WaypointActionListener ActionId confusion

### ISSUE THREAD

**Issue by tkocik92** [dji-sdk/Mobile-SDK-Android-V5#189]
SDK: `5.6.0`
Drone: Mavic 3 Thermal (if it matters)

I am attempting to implement the `WaypointActionListener` in my project. I have implemented a listener similar to the following:

```
final WaypointActionListener actionListener = new WaypointActionListener() {
        @Override
        public void onExecutionStart(int actionId) {
            Timber.tag("TEST").i("Waypoint action %d started", actionId);
        }

        @Override
        public void onExecutionFinish(int actionId, @Nullable IDJIError error) {
            Timber.tag("TEST").i("Waypoint action %d finished", actionId);
            if (error != null) {
                Timber.tag("TEST").e("Waypoint action error %s - %s", error.errorCode(), error.description());
            }
        }
    };
```

After adding the listener and executing a mission, I verified that all of my `Timber` statements are appropriately firing. However, the `actionId` coming back is inconsistent with the documentation.

For example, if Waypoint 0, Action Group 0 (aka the first waypoint and action group) has 2 actions (change gimbal pitch and rotate drone), I would expect my Timber statements to print the following:

```
Waypoint action 0 started // Change gimbal pitch start
Waypoint action 0 finished // Change gimbal pitch finished
Waypoint action 1 started // Rotate drone start
Waypoint action 1 finished // Rotate drone finished
```

But the actual results are:

```
Waypoint action 0 started // Change gimbal pitch start
Waypoint action 0 started // Rotate drone start
Waypoint action 0 finished // One of the above finished
Waypoint action 0 finished // The other finished
```

In other tests, I have found that it is likely that `actionId` is actually the `actionGroupId`.

This raises 2 questions:

1. What is `actionId` supposed to be? According to the documentation, it should be the action id of the action performed.
2. Are waypoint actions supposed to start before the previous one finished?

**Comment by dji-dev** [comment]
Agent comment from Hummels Lei in <a href='https://djisdksupport.zendesk.com/agent/tickets/93676'>Zendesk ticket #93676</a>:<br><br>1. I would like to inform you that we are aware of the issue you have encountered. The correct term for "actionId" is actually "actionGroupId". Rest assured, we are working diligently to resolve this problem in the upcoming version.

 2.  I would like to explain that once the previous waypoint action is completed, the subsequent waypoint action will commence immediately. Due to their close timing, there may be a potential misalignment in the printing process.
<br><p><em><small>°°°</small></em></p>

**Comment by tkocik92** [comment]
Leaving this here for anyone who visits: the problem still exists on `5.7.0`.

**Comment by dji-dev** [comment]
Agent comment from Hummels Lei in <a href='https://djisdksupport.zendesk.com/agent/tickets/93676'>Zendesk ticket #93676</a>:<br><br>I would like to inform you that the issue you have reported is currently planned to be resolved in version 5.8.
<br><p><em><small>°°°</small></em></p>

**Comment by PeimanAtaeiJPDroni** [comment]
> Agent comment from Hummels Lei in [Zendesk ticket #93676](https://djisdksupport.zendesk.com/agent/tickets/93676):I would like to inform you that the issue you have reported is currently planned to be resolved in version 5.8.
> 
> _°°°_

But it is not solved in version 5.8.0-a4

I have the same problem also in this version

**Comment by tkocik92** [comment]
Verified fixed in `5.8.0`
